core/geometry/src/test/topology/RegionOps.test.ts

```diff
@@ -1011,7 +1017,89 @@ describe("RegionOps", () => {
     GeometryCoreTestIO.saveGeometry(allGeometry, "RegionOps", "constructAllXYRegionLoops");
     expect(ck.getNumErrors()).toBe(0);
   });
+
+  it("RegionBooleanMerge", () => {
+    const ck = new Checker();
+    const allGeometry: GeometryQuery[] = [];
+    let isArray = false, hasLoopEntries = false;
+    const arrayConsistsOfLoops = (a: Array<any>) => a.every((value: any) => value instanceof Loop);
+    const regions = IModelJson.Reader.parse(JSON.parse(fs.readFileSync("./src/test/data/curve/areaBoolean/unionRegionWithOverlappingLoops.imjs", "utf8")));
+    if (ck.testDefined(regions, "read regions from file")) {
+      if ((isArray = Array.isArray(regions)) && ck.testTrue(isArray, "regions is an array")) {
```

Comment:
What's the benefit of 
```ts
​ if ((isArray = Array.isArray(regions)) && ck.testTrue(isArray, "regions is an array")) {
```
over
```ts
isArray = Array.isArray(regions);
​if (ck.testTrue(isArray, "regions is an array")) {
```
I know it makes it 1 liner but it's a bit confusing as it looks `isArray = Array.isArray(regions)` is checking a condition while it's just assigning `isArray`

Reply:
Both statements are necessary because 1) I want this test to verify that `regions` is indeed an array, but 2) `ck.testTrue` swallows the `Array.isArray` type guard that suppresses Typescript complaining about using `regions.length` below.  I could split it into two lines, but `if (a = foo(b))` is a common idiom testing for truthiness of `a`, and core-geometry tests are rife with compound if statements that look like `if (ck.testFoo() && type_guard)` because most Checker tests aren't type guards. Ideally we should write a `ck.testArray` that serves as a type guard so all we'd need is `if (ck.testArray(regions, "regions is an array"))` and Typescript would understand that in the subsequent scope, `regions` is an `Array`.